### PR TITLE
Close #12335: Eliminate goto statement in footpath_chain_ride_queue()

### LINKED ISSUE #12335: Remove goto from footpath_chain_ride_queue
https://github.com/OpenRCT2/OpenRCT2/blob/d10b615eb33f6cd597340d83d68eaaa97c1a493e/src/openrct2/world/Footpath.cpp#L1093-L1100

### PR DESCRIPTION
**Pull request by Ron4u**
please verify and make changes wherever possible.
i am not familiar with clang-format. i tried my best to come up with a bug fix.
suggest changes if required

### CODE REVIEW

File: src/openrct2/world/Footpath.cpp [hunk 1]
```diff
@@ -1090,13 +1093,15 @@ void footpath_chain_ride_queue(
...
 
                     baseZ -= LAND_HEIGHT_STEP;
-                    goto foundNextPath;
+                    move = true;
+                    break;
                 }
             } while (!(tileElement++)->IsLastForTile());
         }
+        if(!move)
         break;
```
Comment (tupaschoal) [inline src/openrct2/world/Footpath.cpp:1103]:
```suggestion
        if (!move)
            break;
```

File: src/openrct2/world/Footpath.cpp [hunk 2]
```diff
@@ -1064,6 +1064,7 @@ void footpath_chain_ride_queue(
 
         auto targetQueuePos = curQueuePos + CoordsDirectionDelta[direction];
         tileElement = map_get_first_element_at(targetQueuePos);
+        bool move = false;
```
Comment (Broxzier) [inline src/openrct2/world/Footpath.cpp:1067]: Please use a more descriptive identifier that explains what it means when `move` is true or false. From what I can see from quickly looking at the code, this is used to verify that a queue was found and can be connected.

File: src/openrct2/world/Footpath.cpp (hunk 1 above)
Comment (Ron4u) [reply src/openrct2/world/Footpath.cpp:1103]: Thank you for the suggestion

File: src/openrct2/world/Footpath.cpp (hunk 2 above)
Comment (Ron4u) [reply src/openrct2/world/Footpath.cpp:1067]: Yes, earlier in the code goto statement was used to connect it to queue but there is no such requirement we can just do this by using simple break statements and apply condition checkings at point.

Comment (Broxzier) [reply src/openrct2/world/Footpath.cpp:1067]: If that's the case, then a better identifier would be `foundQueue`, or something similar. Then the statement `if (!foundQueue)` makes immediate sense without having to know how the value was set.

Comment (Ron4u) [reply src/openrct2/world/Footpath.cpp:1067]: Noted. Will create a pr with foundQueue variable. Thank you

Comment (ZehMatt) [reply src/openrct2/world/Footpath.cpp:1067]: This should be changed in this PR not in new one, this PR will not be merged unless we are happy with the code. You can just add more commits on top of this PR.

**Comment by tupaschoal** [comment]
@Ron4u I'm converting this to a draft PR, please tend to the comments and the failures and when it's good to go mark it as ready to review

**Pull request merged**